shell/probe/nocturnal.sh

```diff
@@ -1,30 +1,29 @@
 #!/bin/bash
 
 PRELUDE_API=${PRELUDE_API:="https://api.preludesecurity.com"}
+PRELUDE_DIR=${PRELUDE_DIR:="."}
+PRELUDE_SLEEP=${PRELUDE_SLEEP:=14440}
 
-sys=$(uname -s)-$(uname -m)
-dos=$(echo $sys | tr '[:upper:]' '[:lower:]')
+dos=$(echo $(uname -s)-$(uname -m))
 
 while :
-do
-    temp=$(mktemp)
-    location=$(curl -sL -w %{url_effective} -o $temp -H "token:${PRELUDE_TOKEN}" -H "dos:${dos}" -H "dat:${dat}" $PRELUDE_API)
-    test=$(echo $location | grep -o '[0-9a-f]\{8\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{12\}' | head -n 1)
-
-    if [ -z "$test" ];then
-        sleep 14400
-    else
+do  
+    exe=$PRELUDE_DIR/$(uuidgen).vst
+    location=$(curl -sL -w %{url_effective} -o $exe -H "token:${PRELUDE_TOKEN}" -H "dos:${dos}" -H "dat:${dat}" $PRELUDE_API)
+    
+    if [ -s $exe ];then
+        test=$(echo $location | grep -o '[0-9a-f]\{8\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{12\}' | head -n 1)
         ca=$(echo $location | sed -e 's|^[^/]*//||' -e 's|/.*$||')
-        if [ -z "$PRELUDE_CA" ] || [ "$PRELUDE_CA" == "$ca" ];then
-            chmod +x $temp
 
-            if test -f "$temp";then
-                $temp
-                dat="${test}:$?"
-                $temp -cleanup
-            else
-                dat="${test}:9"
+        if [ -z "$PRELUDE_CA" ] || [ "$PRELUDE_CA" == "$ca" ];then
```

Comment:
the probe will also spin forever (skipping sleep) if this CA check fails

Reply:
ah, because it'll re-request. 
that exists today but I'll see if I can revise that in my change set here.

Reply:
might I recommend something like this in the while
```
if (test) 
    if (ca passes)
         yay run test
         continue
     fi
fi
clean up files
sleep
```